tcsocket/app/worker.py

```diff
@@ -228,9 +230,16 @@ async def delete_old_appointments(ctx):
         logger.info('%d old appointments deleted', v.rowcount)
 
 
+async def kill_worker(ctx):
+    pid = os.getppid()
+    os.kill(os.getppid(), SIGTERM)
+    os.waitpid(pid, 0)
+    logger.info('Killed worker pid %s nightly', pid)
+
+
 class WorkerSettings:
     functions = [get_image, submit_booking, submit_enquiry, update_contractors, update_enquiry_options]
-    cron_jobs = [cron(delete_old_appointments, hour={0, 3, 6, 9, 12, 15, 18, 21})]
+    cron_jobs = [cron(delete_old_appointments, hour={0, 3, 6, 9, 12, 15, 18, 21}), cron(kill_worker, hour=3)]
```

Comment:
Maybe clearer to use `cron` as a decorator.

Reply:
Do you have cron as a decorator? Or are you asking us to build a decorator? I can't see one in your docs/code.